### PR TITLE
Merchant items

### PR DESCRIPTION
**Pull request by GregJMitchell**
This Branch Does:
- Create a merchant item index page
- Add deactivate and activate buttons that can change an items status
- Creates a form that allows a merchant user to create a new item
- Allows a merchant user to delete an item if it has never been ordered
- Allows a merchant user to update/edit an item
All tests passing

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -41,12 +41,9 @@
 
   namespace :merchant do
     get "/", to: "dashboard#show"
-    get "/items", to: "items#index"
-    get "/items/new", to: "items#new"
-    post "/items", to: "items#create"
     post "/items/deactivate", to: "items#deactivate"
     post "/items/activate", to: "items#activate"
-    delete "/items/", to: "items#destroy"
+    resources :items, except: [:put]
```
Comment (abcdefghijohn) [inline config/routes.rb:46]: Awesome use of resources here, cleaned up routes quite nicely

**Review by abcdefghijohn — APPROVED** [review]
Looks great!

File: app/controllers/merchant/items_controller.rb [hunk 2]
```diff
@@ -1,5 +1,70 @@
 class Merchant::ItemsController < Merchant::BaseController
   def index
-    @items = Merchant.find(params["merchant_id"]).items
+    user = User.find(session[:user_id])
+    @items = Merchant.find(user.merchant.id).items
+    @merchant = Merchant.find(user.merchant.id)
```
Comment (jakeheft) [inline app/controllers/merchant/items_controller.rb]: We shouldn't be setting 2 instance variables in a controller action. Proper convention is to set one and us that inside your view/use models methods to get what you need. I would keep @merchant, then use @merchant.items to get items.

File: app/controllers/merchant/items_controller.rb [hunk 3]
```diff
@@ -1,5 +1,70 @@
...
+    @item = Item.new
+  end
+
+  def create
+    @item = Item.new(item_params)
+    if @item.save
+      redirect_to merchant_items_path
+      flash[:notice] = "New item has been created"
+    else
+      flash[:error] = @item.errors.full_messages.to_sentence
```
Comment (jakeheft) [inline app/controllers/merchant/items_controller.rb:17]: Nice use of the error messages

File: app/controllers/merchant/items_controller.rb [hunk 4]
```diff
@@ -1,5 +1,70 @@
...
+  end
+
+  def deactivate
+    item = Item.find(params[:item_id])
+    item.update(active?: false)
+    redirect_to merchant_items_path
+    flash[:notice] = "This item is no longer for sale"
+  
+  end
+  def activate
```
Comment (jakeheft) [inline app/controllers/merchant/items_controller.rb]: Need to move line space

File: app/views/merchant/items/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  <%= f.label :price %>
+  <%= f.text_field :price %>
+
+  <%= f.label :inventory %>
+  <%= f.text_field :inventory %>
+
+  <%= f.hidden_field :merchant_id , :value => current_user.merchant.id %>
+
+  <%= f.submit %>
+<% end %>
```
Comment (jakeheft) [inline app/views/merchant/items/new.html.erb:20]: Nice form_with!

File: db/migrate/20190817190207_create_items.rb [hunk 6]
```diff
@@ -4,7 +4,8 @@ def change
       t.string :name
       t.string :description
       t.integer :price
-      t.string :image
+      t.string :image, :default =>  "https://www.monsterchildren.com/wp-content/uploads/2020/02/nic-cage-monster-children-1068x571.jpg"
```
Comment (jakeheft) [inline db/migrate/20190817190207_create_items.rb:7]: lmao

**Review by jakeheft — CHANGES_REQUESTED** [review]
Good work with your user stories. Once the requested changes have been made, everything should be good to merge.

**Review by GarrettCottrell — APPROVED** [review]

**Review by jakeheft — APPROVED** [review]

**Pull request merged**